### DISCUSSION TITLE
Removing codeowner review requests automatically when the diff changes

### DISCUSSION THREAD

**Discussion by Hari-07** [community/community#84184] [Pull Requests]
### Select Topic Area

Product Feedback

### Body

When changes are pushed up to a branch which has a PR open, it automatically triggers review requests from everyone who is associated with any of the files in the diff.

However if this is changed, and fewer files are now changed, codeowners associated with the files that are no longer in the diff continue to be in the review requested category.

This is especially an issue if someone pushes up a merge commit, instead of rebasing, and many authors having nothing to do with the PR get assigned. The diff gets sorted by force pushing, but the review requests remain

**Comment by bryan-shea** [comment]
In GitHub, when updates to a PR branch change the diff (such as removing files), the automatically triggered review requests to code owners of the now-irrelevant files persist.
- **Specific Concern:** This becomes problematic with merge commits, leading to unnecessary review requests that don't automatically clear even after correcting the diff.

### Solutions and Insights:

1. **Manual Reviewer Management:**
   - **Current GitHub Limitation:** GitHub does not automatically update review requests based on diff changes.
   - **Action Required:** PR authors or maintainers need to manually remove irrelevant reviewers when the diff changes.

2. **Automated Reviewer Adjustment via GitHub Actions:**
   - **Automated Solution:** Implement a GitHub Action to handle review requests dynamically.
   - **Trigger:** Set the action to activate on push events to PR branches.
   - **Process:** The action analyzes the diff, identifies relevant code owners, and updates the review requests to reflect the current state of the diff.
   - **GitHub API Usage:** Utilize GitHub's API within the action for managing review requests.

3. **Fallback to Manual Review:**
   - **Complex Scenarios:** In cases of complicated merges or sophisticated code ownership setups, a combination of automated and manual review management might be necessary.

While GitHub does not currently offer an automatic solution for updating review requests based on diff changes, a blend of manual management and custom automation using GitHub Actions can significantly streamline the process. This ensures that review requests remain relevant to the current state of the PR.

**Comment by swrobel** [comment]
Seems like a dupe of https://github.com/orgs/community/discussions/66936
